src/wallet/gtest/test_rpc_wallet.cpp

```diff
@@ -239,3 +270,137 @@ TEST(WalletRPCTests, RPCZsendmanyTaddrToSapling)
     RegtestDeactivateSapling();
     UnloadGlobalWallet();
 }
+
+TEST(WalletRPCTests, ZIP317Fee)
+{
+    LoadProofParameters();
+    SelectParams(CBaseChainParams::TESTNET);
+
+    LoadGlobalWallet();
+
+    RegtestActivateSapling();
+    {
+        LOCK2(cs_main, pwalletMain->cs_wallet);
+
+        if (!pwalletMain->HaveMnemonicSeed()) {
+            pwalletMain->GenerateNewSeed();
+        }
+
+        KeyIO keyIO(Params());
+        // add keys manually
+        auto taddr = pwalletMain->GenerateNewKey(true).GetID();
+        auto pa = pwalletMain->GenerateNewLegacySaplingZKey();
+
+        const Consensus::Params& consensusParams = Params().GetConsensus();
+
+        int nextBlockHeight = chainActive.Height() + 1;
+
+        // Add a fake transaction to the wallet
+        CMutableTransaction mtx = CreateNewContextualCMutableTransaction(consensusParams, nextBlockHeight, false);
+        CScript scriptPubKey = CScript() << OP_DUP << OP_HASH160 << ToByteVector(taddr) << OP_EQUALVERIFY << OP_CHECKSIG;
+        size_t utxoCount = 100;
+        for (size_t i = 0; i < utxoCount; i++) {
+            mtx.vout.push_back(CTxOut(5 * COIN, scriptPubKey));
```

Comment:
```suggestion
            mtx.vout.emplace_back(5 * COIN, scriptPubKey);
```

Reply:
Did this get reverted when fixing the conflict?

Reply:
Oh, it did – I hadn’t pulled that commit when I rebased on master, then force-pushed 😞